pango/src/attribute.rs

```diff
@@ -200,4 +204,200 @@ impl Attribute {
             (*stash.0).end_index = index;
         }
     }
+
+    pub fn downcast<T: IsAttribute>(self) -> Result<T, Attribute> {
+        unsafe {
+            if T::ATTR_TYPES.contains(&self.get_attr_class().type_()) {
+                Ok(from_glib_full(self.to_glib_full()))
+            } else {
+                Err(self)
+            }
+        }
+    }
+
+    pub fn downcast_ref<T: IsAttribute>(&self) -> Option<&T> {
+        unsafe {
+            if T::ATTR_TYPES.contains(&self.get_attr_class().type_()) {
+                Some(&*(self as *const Attribute as *const T))
+            } else {
+                None
+            }
+        }
+    }
+}
+
+pub trait IsAttribute:
```

Comment:
```suggestion
pub unsafe trait IsAttribute:
```

Reply:
What makes it need unsafe?  I'm following the examples of `ParamSpec` and `RenderNode`.  They both have a similar type and don't need unsafe.

Reply:
I created issues for marking those other two traits as unsafe. Can you update this PR accordingly? Thanks :)